app.js

```diff
@@ -0,0 +1,168 @@
+const favoriteBar = document.getElementById('favorite-container');
+const cardContainer = document.getElementById('card-container');
+const nextPage = document.getElementById('next-page');
+const prevPage = document.getElementById('previous-page');
+searchInput.addEventListener('search', search);
+nextPage.addEventListener('click', page);
+prevPage.addEventListener('click', page);
+
+load();
+
+// This function is used to boot the app. 
+// It renders the favorites and the card grid.
+async function load() {
+    if (localStorage.getItem('favorites')) {
+        await getData(localStorage.getItem('favorites'), 'faves');
+        await showData('faves');
+    }
+    await getData(`?page=${currentPage}`, 'grid');
+    await showData('grid');
+}
+
+// This function is used to change the page. 
+// It gets new char content and renders it in the grid.
+async function page(event) {
+    console.log(currentPage);
+    targetID = event.target.id;
+    currentPage = targetID === 'next-page' ? currentPage + 1 : currentPage - 1;
+    if (currentPage < 1) { currentPage = 1; }
+    await getData(`?page=${currentPage}`, 'grid');
+    await showData('grid');
+}
+
+// This function is used to search for characters of a given name.
+// It gets char info and renders it to the grid. 
+async function search(event) {
+    const searchTerm = event.target.value.toUpperCase();
+    await getData(`?name=${searchTerm}`, 'grid');
+    await showData('grid');
+}
+
+// Called on heart click.
+async function favorite(event) {
+    if (!event.target.classList.contains('favoriteButton')) {
+        return;
+    }
+    
+    toggleHeartDisplay(event);
+
+    const favorites = localStorage.getItem('favorites');
+    const favoriteArray = favorites ? favorites.split(',') : [];
+
+    // Removing a favorite
+    if (favoriteArray.includes(event.currentTarget.id)) {
+        favoriteArray.splice(favoriteArray.indexOf(event.currentTarget.id), 1);
+    } else {
+        // Add a favorite
+        favoriteArray.push(event.currentTarget.id);
+    }
+
+    localStorage.setItem('favorites', favoriteArray);
+    await getData(localStorage.getItem('favorites'), 'faves');
+    await showData('faves');
+}
+
+async function getData(queryParam, destination) {
+    if (queryParam === '' && destination === 'faves') {
+        favoriteCharactersData = [];
+        return;
+    }
+
+    const data = await fetch(`${RickAndMortyAPI}${queryParam}`);
+    const parsedResponse = await data.json();
+
+    // check out data.info for metadata
+    if (destination === 'grid') {
+        characterGridData = parsedResponse.results;
+    } else {
+        favoriteCharactersData = Array.isArray(parsedResponse) ? parsedResponse : [parsedResponse];
+    }
+}
+
+// Looks at characterGridData and renders it in the grid.
+function showData(destination) {
+    if (destination === 'grid') {
+        cardContainer.innerHTML = '';
+    } else {
+        favoriteBar.innerHTML = '';
+    }
```

Comment:
It _might_ be good to have a helper method for this. 

We could also entertain the idea of hiding and showing DOM elements instead of removing them entirely.